src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseIdentityReport.java

```diff
@@ -0,0 +1,233 @@
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.ParseException;
+
+public class PreciseIdentityReport extends TableReport {
+  public static final String REPORT_NAME = "preciseidentity";
+  public static final String CATEGORY = REPORT_CATEGORY_COUNTS;
+  private String start;
+  private String end;
+
+  private static final Option OPT_AFTER = CommonOptions.after(true);
+  private static final Option OPT_BEFORE = CommonOptions.before(true);
+
+  List<List<String>> table;
+  List<SampleDto> allPreciseIdentities;
+
+  private enum TimePoint {
+    // Coded by the Times Received portion of the tissue name: PRE_1111111_Xx_X_nn_*#*-1
+    RANDOMIZATION("Screening/Randomization", 1),
+    SIX("6 mo Follow Up", 2),
+    TWELVE("1 yr Follow Up", 3), //
+    EIGHTEEN("18 mo Follow Up", 4),
+    TWENTY_FOUR("2 yr Follow Up", 5),
+    BX("Bx Slides", 6),
+    RP("RP Slides", 7);
+    private final String key;
+    private final int timePointCode;
+    private static final Map<String, PreciseIdentityReport.TimePoint> lookup = new TreeMap<>();
+
+    static {
+      for (PreciseIdentityReport.TimePoint s :
+          EnumSet.allOf(PreciseIdentityReport.TimePoint.class)) {
+        lookup.put(s.getKey(), s);
+      }
+    }
+
+    TimePoint(String key, int timePointCode) {
+      this.key = key;
+      this.timePointCode = timePointCode;
+    }
+
+    public String getKey() {
+      return key;
+    }
+
+    /**
+     * Assumes that PRECISE samples will always have _nn_ in the name, and that Tissue Origin will
+     * never be "nn". Should hopefully be a safe assumption since this is a biobanking project where
+     * tracking tissue origin is critical.
+     */
+    public Predicate<SampleDto> predicate() {
+      return sample -> {
+        if (SAMPLE_CATEGORY_IDENTITY.equals(sample.getSampleType())) return false;
+        return timePointCode == getTimesReceived(sample.getName());
+      };
+    }
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException, IOException {
+    List<SampleDto> allPreciseSamples =
+        pinery
+            .getSample()
+            .allFiltered(new SampleClient.SamplesFilter().withProjects(Lists.newArrayList("PRE")));
+    List<String> tissueOriginOrder = Lists.newArrayList("_Se_", "_Pl_", "_Ly_");
+    List<String> tissueOriginOrderExtra =
+        Lists.newArrayList("_Se_", "_Pl_", "_Ly_", "_Us_", "_Up_");
+    allPreciseIdentities =
+        allPreciseSamples
+            .stream()
+            .filter(s -> s.getSampleType().equals(SAMPLE_CATEGORY_IDENTITY))
+            .collect(Collectors.toList());
+
+    table = new LinkedList<>();
+
+    // Add headers
+    table.add(
+        Lists.newArrayList(
+            "ID",
+            "Screening/Randomization Serum",
```

Comment:
Since the columns never change, they should be defined statically and returned from `getColumns()`. See `ProjectSequencingReport` for a simple example

Reply:
Updated